pkg/cmd/delete.go

```diff
@@ -120,3 +127,53 @@ func (command *deleteCmdOptions) run(args []string) error {
 
 	return nil
 }
+
+func getIntegration(ctx context.Context, c client.Client, name string, namespace string) (*v1.Integration, error) {
+	key := k8sclient.ObjectKey{
+		Name:      name,
+		Namespace: namespace,
+	}
+	answer := v1.NewIntegration(namespace, name)
+	if err := c.Get(ctx, key, &answer); err != nil {
+		return nil, err
+	}
+	return &answer, nil
+}
+
+func deleteIntegration(ctx context.Context, c client.Client, integration *v1.Integration) error {
+	deleted, binding, err := deleteKameletBindingIfExists(ctx, c, integration)
+	if err != nil {
+		return err
+	}
+	if deleted {
+		// Deleting KameletBinding will automatically clean up the integration
+		fmt.Println("KameletBinding " + binding + " deleted")
+		return nil
+	}
+	return c.Delete(ctx, integration)
+}
+
+func deleteKameletBindingIfExists(ctx context.Context, c client.Client, integration *v1.Integration) (bool, string, error) {
+	kind := integration.ObjectMeta.GetLabels()[kubernetes.CamelCreatorLabelKind]
```

Comment:
It's a good idea to add those labels to have better visibility, however we may also check the metadata->ownerReferences of the Integration to find the related KameletBinding.

Reply:
OK, I'll add double checking with ownerReferences here for finding a KLB.